packages/oui-button/src/button.controller.js

```diff
@@ -0,0 +1,69 @@
+import _ from 'lodash'
+
+export default class {
+  constructor ($attrs, $element, $log) {
+    'ngInject'
+
+    this.$attrs = $attrs
+    this.$element = $element
+    this.$log = $log
+  }
+
+  $onInit () {
+    this.checkModifiers()
+    this.checkSteps()
+    this.checkType()
+  }
+
+  $postLink () {
+    // Remove ID and Name to avoid duplicate
+    this.$element.removeAttr('id')
+    this.$element.removeAttr('name')
+  }
+
+  // Check presence of attributes for modifiers
+  checkModifiers () {
+    this.primary = this.$attrs.hasOwnProperty('primary')
+    this.secondary = this.$attrs.hasOwnProperty('secondary')
+    this.link = this.$attrs.hasOwnProperty('link')
+
+    // Warning for multiple declaration
+    this.checkAttrs([
```

Comment:
Instead to pass two time attributes names maybe we could just call it with this signature `checkAttrs('primary', 'secondary', 'link')`. With this signature we could also do the `this.$attrs.hasOwnProperty(attributeName)` in this method.

Also, maybe `checkAttrs` is not best name because it doesn't specify what is the check.

So, could it be better to do this method:

```javascript
createOrAttrs (...attributes) {
  attributes.forEach((attributeName) => {
     this[attributeName] = this.$attrs.hasOwnProperty(attributeName)
  });

  if (_.sum(_.values(_.pick(this, attributes))) > 1) {
     var formattedAttributes = attributes.map((attributeName) => _.kebabCase(attributeName)).join('|')

     this.$log.warn(`Unauthorized multiple declaration of attributes (${formattedAttributes})`)
  }
}
```

This method should also be extracted in a service since not just the button will have this kind of attributes.

Reply:
it looks too complicated to me, maybe something simpler like :
```
if (this.primary + this.secondary + this.link === 1) {
  // good
} else {
 // bad
}
```

Reply:
my bad you did the same thin with sum

Reply:
Yes, good idea. But we should add this on a service. Can be useful for the others components.